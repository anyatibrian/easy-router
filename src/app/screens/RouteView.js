import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'
import { RoutingPoints } from '../actions/routeActions/'
import MapsView from '../components/MapsView'

export class RouteView extends Component {
    constructor(props){
        super(props)
        this.state={
            userLatitude:'',
            userLongitude:'',
        }
    }
    
    /**
     * handlingcyclingRoutes
     */
    handleRoutes=(mode)=>{
        const{RoutingPoints, route:{params:{originInfo, destinationInfo}}} = this.props
        RoutingPoints(originInfo,destinationInfo, mode)
    }
    componentDidMount(){
        const { originInfo, destinationInfo} =this.props.route.params
        const { RoutingPoints }= this.props
        RoutingPoints(originInfo, destinationInfo, 'car')
    }
    render() {
        const {userLatitude, userLongitude} = this.state
        const { coordinates, loading, route:{params:{originInfo, destinationInfo}}} = this.props
        return (
            <View>
                <MapsView coordinates={coordinates}
                userLatitude={userLatitude}
                userLongitude={userLongitude}
                routingPoint={RoutingPoints}
                handleRoutes={this.handleRoutes}
                destination={destinationInfo.destination}
                origin={originInfo.origin}
                loading={loading}
                />
            </View>
        )
    }
}

export const mapStateToProps = state=>({
    coordinates: state.routResult.coordinatePoints,
    loading:state.routResult.isLoading
  })
export default  connect(mapStateToProps, { RoutingPoints })(RouteView)
