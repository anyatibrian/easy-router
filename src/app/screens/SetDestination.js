import React, { Component } from 'react'
import { 
    View,
    StyleSheet,
    Text,
    ImageBackground,
    TouchableOpacity
  } from 'react-native'
import { connect } from 'react-redux'
import FormInputs from '../components/FormInputs'
import { routeAction } from '../actions/routeActions'
import ResultList from '../components/ResultList'
import taxiImage from '../assets/taxi.jpg'

export class SetDestination extends Component {
    constructor(props){
        super(props)
        this.state ={
            destination:'',
            showList:true,
            destinationInput:{
              type:'textinput',
              value:null,
              lat:'',
              lng:''
            }
        }
    }

/**
 * @param {name}
 * @param {value}
 * 
 */
 
 handleTextChange=(value)=>{
    this.setState({showList:true})
    const { routeAction } = this.props
    //making call to the route actions
    routeAction(value)
  }
 
  /**
   * navigation to the mapsScreen passing the params 
   */
  goViewMapScreen = () =>{
      const {destination, destinationInput} = this.state
      const { route:{
                    params:{origin,lat,lng}
                },
                navigation:{
                    navigate
                } } = this.props
      navigate("RouteView", {
          originInfo:{
              lat:lat,
              lng:lng
          },
          destinationInfo:{
              lat:destinationInput.lat,
              lng:destinationInput.lng
          }
      })
      
  }

  /**
   * set the current location on the map
   * 
   */
 
  setDestination=(results, location)=>{
    //making various declarations 
     this.setState({destination:results,showList:false, destinationInput:{
       type:'textinput',
       value:null,
       lat:location.lat,
       lng:location.lng
     }})
  }
 
    render() {
        const {destination, destinationInput,showList} = this.state
        const {searchResult, loading}=this.props
         const displayResult = searchResult.map(result=>
         <ResultList 
         results={result} 
         key={result.geometry.lat}
         setDestination={this.setDestination}
         loading={loading}
        />
    )
        return (
            <View>
               <View style={styles.container}>
                <ImageBackground  source={taxiImage} style={styles.image}>
                <Text style={styles.text}>Where do you want to go ?</Text>
                   <FormInputs
                    style={styles.textInput}
                    placeholder="set your current location..."
                    autoCapitalize="none"
                    type={destinationInput.type}
                    value={destination}
                    placeholderTextColor="white"
                    onChangeText={value => {
                      this.handleTextChange( value)
                      this.setState({destination:value})
                    }}
                  />
                  { showList? displayResult: null }
                    <TouchableOpacity style={styles.Button} onPress={()=>this.goViewMapScreen()}>
                      <Text style={styles.buttonText}> View Routes</Text>
                    </TouchableOpacity>
              </ImageBackground>
            </View>
            </View>
          )
        }
      }
      
      export const mapStateToProps = state=>({
        searchResult: state.routResult.searchResult,
        loading:state.routResult.isLoading
      })
      
      // export the component 
      export default connect(mapStateToProps, {routeAction})(SetDestination)
      
      //styles sheet for the android application
      const styles = StyleSheet.create({
        container:{
          height:'100%'
        },
        textInput:{
          borderRadius:10,
          borderColor:'white',
          borderWidth:2,
          padding:10,
          marginLeft:15,
          marginRight: 10,
          marginBottom:5,
          fontSize:18,
          color:'white'
        },
        text:{
          fontSize:25,
          fontWeight:'bold',
          color:'white',
          padding:30,
          marginTop: 200
      
        },
        image:{
          resizeMode:'cover',
          height:'100%',
        },
        Button:{
          backgroundColor: 'white',
          marginLeft:15,
          marginRight: 10,
          marginTop: 15,
          padding:15
      
        },
        buttonText:{
          fontSize: 20,
          color:'#f57b51',
          marginLeft:100
        }
      })
      