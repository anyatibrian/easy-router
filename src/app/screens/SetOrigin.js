import React, { Component } from 'react'
import {
  View,
  StyleSheet,
  PermissionsAndroid,
  Platform,
  Text,
  ImageBackground,
  TouchableOpacity, 
} from 'react-native'
import { connect} from 'react-redux'
import FormInputs from '../components/FormInputs'
import {routeAction} from '../actions/routeActions'
import ResultList from '../components/ResultList'
import taxiImage from '../assets/taxi.jpg'

export class SetOrigin extends Component {
  constructor(props){
    super(props)
    this.state ={
      origin:'',
      showList:true,
      originInput:{
        type:'textinput',
        value:null,
        lat:'',
        lng:''
      }
  }
}

/**
 * @param {name}
 * @param {value}
 */
 
 handleTextChange=(value)=>{
   this.setState({showList:true})
   const { routeAction } = this.props
   //making call to the route actions
   routeAction(value)
 }

 gotoDestinationScreen = () =>{
   const { origin, originInput } = this.state
   this.props.navigation.navigate("SetDestination",{
     lat: originInput.lat,
     lng: originInput.lng,
     origin: origin
   })
 }
 /**
  * set the current location on the map
  * 
  */

 setDestination=(results, location)=>{
   //making various declarations 
    this.setState({origin:results,showList:false, originInput:{
      type:'textinput',
      value:null,
      lat:location.lat,
      lng:location.lng
    }})
 }

 /**
  * the component did mount elements
  */
  async componentDidMount(){
     await this.allowMapToAccessLocation()
  }
  /*ask for user permissions to access their
    particular localtion 
  */

  allowMapToAccessLocation = async()=>{
    try{
      if(Platform.OS === 'ios'){
        return true
      }else{
        const granted = PermissionsAndroid.request(
          PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
        if( granted === PermissionsAndroid.RESULTS.GRANTED){
          return true
        }else{
          return false
        }
      }
    }catch(error){
      console.warn(error)
    }
  }

  render() {
    const {
        originInput,
        showList,
        origin,
    } = this.state
    const { searchResult, loading }=this.props
    const displayResult = searchResult.map(result=>
        <ResultList 
        results={result} 
        key={result.geometry.lat}
        setDestination={this.setDestination}
        loading={loading}
        />
    )
    return (
      <View>
         <View style={styles.container}>
          <ImageBackground  source={taxiImage} style={styles.image}>
          <Text style={styles.text}>Where are you located at ?</Text>
             <FormInputs
              style={styles.textInput}
              placeholder="set your current location..."
              autoCapitalize="none"
              type={originInput.type}
              value={origin}
              placeholderTextColor="white"
              onChangeText={value => {
                this.handleTextChange( value)
                this.setState({origin:value})
              }}
            />
            { showList? displayResult: null }
              <TouchableOpacity style={styles.Button} onPress={()=>this.gotoDestinationScreen()}>
                <Text style={styles.buttonText}> Set Destination</Text>
              </TouchableOpacity>
        </ImageBackground>
      </View>
      </View>
    )
  }
}

export const mapStateToProps = state=>({
  searchResult: state.routResult.searchResult,
  loading:state.routResult.isLoading
})

// export the component 
export default connect(mapStateToProps, {routeAction})(SetOrigin)

//styles sheet for the android application
const styles = StyleSheet.create({
  container:{
    height:'100%'
  },
  textInput:{
    borderRadius:10,
    borderColor:'white',
    borderWidth:2,
    padding:10,
    marginLeft:15,
    marginRight: 10,
    marginBottom:5,
    fontSize:18,
    color:'white'
  },
  text:{
    fontSize:25,
    fontWeight:'bold',
    color:'white',
    padding:30,
    marginTop: 200

  },
  image:{
    resizeMode:'cover',
    height:'100%',
  },
  Button:{
    backgroundColor: 'white',
    marginLeft:15,
    marginRight: 10,
    marginTop: 15,
    padding:15

  },
  buttonText:{
    fontSize: 20,
    color:'#f57b51',
    marginLeft:100
  }
})
