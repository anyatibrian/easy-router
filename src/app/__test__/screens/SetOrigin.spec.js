import React from 'react'
import {shallow } from 'enzyme'
import {PermissionsAndroid} from 'react-native'
import { SetOrigin, mapStateToProps } from '../../screens/SetOrigin'

const props ={
    searchResult:[], 
    loading:false,
    routeAction:jest.fn(),
    navigation:{
        navigate:jest.fn()
    },
    handleTextChange: jest.fn()
}
let instance
let wrapper

describe('Should test set Origin', () => {
    beforeEach(()=>{
        wrapper = shallow(<SetOrigin {...props} />)
        instance = wrapper.instance()
    })

    it( 'should the test the component mount successfully', ()=>{
        expect(wrapper).toMatchSnapshot()
    })

    it('should test the handleChangeMethod method',()=>{
        instance.handleTextChange('kampala')
        wrapper.setState({showList:true})
        expect(props.routeAction).toHaveBeenCalled()
        expect(instance.state.showList).toEqual(true)
    })

    it('should test go to destination screen',()=>{
        instance.gotoDestinationScreen()
        expect(props.navigation.navigate).toHaveBeenCalled()
    })

    it('should test setDestination method',()=>{
        const location={
            lat:32.542834,
            lng:0.210184
        }
        instance.setDestination('kampala', location )
        wrapper.setState({
            destination:'kampala',
            showList:false,
            destinationInput:{
                type:'textinput',
                value:null,
                lat:32.542834,
                lng:0.210184
              }
    })

    expect(instance.state.destination).toEqual('kampala')
    })

    it('should test maps state to props', ()=>{
        const state = {
            routResult:{
                searchResult:[],
                loading:undefined
            }
        }
        expect(mapStateToProps(state))
    })
    it('should test onchange event handler',()=>{
        const formInput = wrapper.find('FormInputs').first()
        formInput.props().onChangeText()
        const handleChange = jest.spyOn(instance,'handleTextChange')
        expect(handleChange).toHaveBeenCalledTimes(0)
    })
})
