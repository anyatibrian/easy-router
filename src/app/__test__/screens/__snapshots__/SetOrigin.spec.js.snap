// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test set Origin should the test the component mount successfully 1`] = `ShallowWrapper {}`;
